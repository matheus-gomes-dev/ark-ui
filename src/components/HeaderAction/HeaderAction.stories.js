import React from 'react';

import HeaderAction from './';

const Template = (args) => <HeaderAction {...args} />;

export const ListWallets = Template.bind({});
ListWallets.args = {
  action: 'list-wallets',
};

export const ImportWallet = Template.bind({});
ImportWallet.args = {
  action: 'import-wallet',
};

export const CreateWallet = Template.bind({});
CreateWallet.args = {
  action: 'create-wallet',
};

export const SeeDelegates = Template.bind({});
SeeDelegates.args = {
  action: 'see-delegates',
};

const stories = {
  title: 'Header/Action',
  component: HeaderAction
};
export default stories;
