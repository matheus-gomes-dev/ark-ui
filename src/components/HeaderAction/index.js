import React from 'react';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faWallet, faDownload, faPlus } from '@fortawesome/free-solid-svg-icons';

const icons = {
  'list-wallets': faWallet,
  'import-wallet': faDownload,
  'add-wallet': faPlus
};

const defineIconClasses = (isActive, hasDivider) => {
  const classes = `
    box-border
    h-12
    w-12
    md:h-16
    md:w-24
    cursor-pointer
    ${isActive ? 'text-gray-800' : 'text-gray-600'}
    hover:text-gray-800
    flex
    flex-col
    justify-center
    items-center
  `;
  return hasDivider ? `${classes} border-gray-200 border-r` : classes;

}

const HeaderAction = ({ action, label, isActive = false, hasDivider }) => (
  <div className={defineIconClasses(isActive, hasDivider)}>
    <FontAwesomeIcon className={`${isActive ? 'text-red-600' : ''} text-xl`} icon={icons[action]} title={label}/>
    <div className="hidden lg:block ">
      <span className="text-xs">{label}</span>
    </div>
  </div>
);


HeaderAction.propTypes = {
  action: PropTypes.oneOf(['list-wallets', 'import-wallet', 'add-wallet']),
  label: PropTypes.string,
  isActive: PropTypes.bool,
  hasDivider: PropTypes.bool
};

export default HeaderAction;
