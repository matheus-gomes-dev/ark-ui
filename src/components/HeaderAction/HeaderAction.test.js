import React from 'react';
import renderer from 'react-test-renderer';

import HeaderAction from './';

describe('HeaderAction', () => {
  
  it('should render list wallets component correctly', () => {
    const tree = renderer.create(<HeaderAction action="list-wallets" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render import wallet component correctly', () => {
    const tree = renderer.create(<HeaderAction action="import-wallet" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render add wallet component correctly', () => {
    const tree = renderer.create(<HeaderAction action="add-wallet" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
