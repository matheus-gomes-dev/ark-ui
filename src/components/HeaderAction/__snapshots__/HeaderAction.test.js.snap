// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderAction should render add wallet component correctly 1`] = `
<div
  className="
    box-border
    h-12
    w-12
    md:h-16
    md:w-24
    cursor-pointer
    text-gray-600
    hover:text-gray-800
    flex
    flex-col
    justify-center
    items-center
  "
>
  <svg
    aria-labelledby="svg-inline--fa-title-8TEv8jExLpsO"
    className="svg-inline--fa fa-plus fa-w-14  text-xl"
    data-icon="plus"
    data-prefix="fas"
    role="img"
    style={Object {}}
    viewBox="0 0 448 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title
      id="svg-inline--fa-title-8TEv8jExLpsO"
      style={Object {}}
    >
      Create Wallet
    </title>
    <path
      d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  <div
    className="hidden lg:block "
  >
    <span
      className="text-xs"
    >
      Create Wallet
    </span>
  </div>
</div>
`;

exports[`HeaderAction should render import wallet component correctly 1`] = `
<div
  className="
    box-border
    h-12
    w-12
    md:h-16
    md:w-24
    cursor-pointer
    text-gray-600
    hover:text-gray-800
    flex
    flex-col
    justify-center
    items-center
  "
>
  <svg
    aria-labelledby="svg-inline--fa-title-5ynEigem3KVk"
    className="svg-inline--fa fa-download fa-w-16  text-xl"
    data-icon="download"
    data-prefix="fas"
    role="img"
    style={Object {}}
    viewBox="0 0 512 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title
      id="svg-inline--fa-title-5ynEigem3KVk"
      style={Object {}}
    >
      Import Wallet
    </title>
    <path
      d="M216 0h80c13.3 0 24 10.7 24 24v168h87.7c17.8 0 26.7 21.5 14.1 34.1L269.7 378.3c-7.5 7.5-19.8 7.5-27.3 0L90.1 226.1c-12.6-12.6-3.7-34.1 14.1-34.1H192V24c0-13.3 10.7-24 24-24zm296 376v112c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V376c0-13.3 10.7-24 24-24h146.7l49 49c20.1 20.1 52.5 20.1 72.6 0l49-49H488c13.3 0 24 10.7 24 24zm-124 88c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20zm64 0c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  <div
    className="hidden lg:block "
  >
    <span
      className="text-xs"
    >
      Import Wallet
    </span>
  </div>
</div>
`;

exports[`HeaderAction should render list wallets component correctly 1`] = `
<div
  className="
    box-border
    h-12
    w-12
    md:h-16
    md:w-24
    cursor-pointer
    text-gray-600
    hover:text-gray-800
    flex
    flex-col
    justify-center
    items-center
  "
>
  <svg
    aria-labelledby="svg-inline--fa-title-KDNROGU2wLbQ"
    className="svg-inline--fa fa-wallet fa-w-16  text-xl"
    data-icon="wallet"
    data-prefix="fas"
    role="img"
    style={Object {}}
    viewBox="0 0 512 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title
      id="svg-inline--fa-title-KDNROGU2wLbQ"
      style={Object {}}
    >
      My Wallets
    </title>
    <path
      d="M461.2 128H80c-8.84 0-16-7.16-16-16s7.16-16 16-16h384c8.84 0 16-7.16 16-16 0-26.51-21.49-48-48-48H64C28.65 32 0 60.65 0 96v320c0 35.35 28.65 64 64 64h397.2c28.02 0 50.8-21.53 50.8-48V176c0-26.47-22.78-48-50.8-48zM416 336c-17.67 0-32-14.33-32-32s14.33-32 32-32 32 14.33 32 32-14.33 32-32 32z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  <div
    className="hidden lg:block "
  >
    <span
      className="text-xs"
    >
      My Wallets
    </span>
  </div>
</div>
`;
